admin-dev/themes/new-theme/js/pages/product/components/generator/AttributesSelector.vue

```diff
@@ -119,32 +119,58 @@
   </div>
 </template>
 
-<script>
+<script lang="ts">
+  import {PropType} from 'vue';
   import isSelected from '@pages/product/mixins/is-attribute-selected';
   import ProductMap from '@pages/product/product-map';
   import PerfectScrollbar from 'perfect-scrollbar';
+  // @ts-ignore
   import Bloodhound from 'typeahead.js';
-  import AutoCompleteSearch from '@components/auto-complete-search';
+  import AutoCompleteSearch, {AutoCompleteSearchConfig} from '@components/auto-complete-search';
   import Tokenizers from '@components/bloodhound/tokenizers';
 
   const {$} = window;
 
   const CombinationsMap = ProductMap.combinations;
 
-  export default {
+  export interface AttributesSelectorStates {
+    dataSetConfig: AutoCompleteSearchConfig | {};
+    searchSource: Record<string, any>;
+    scrollbar: PerfectScrollbar | null;
+    hasGeneratedCombinations: boolean;
+    checkboxList: Array<Record<string, any>>;
+  }
+
+  export interface AttributeGroup {
+    id: number;
+    name: string;
+    publicName: string;
+    attributes: Array<Attribute>;
+  }
+
+  /* eslint-disable camelcase */
+  export interface Attribute {
+    id: number;
+    color: string;
+    group_id: number;
+    group_name: string;
+    name: string;
+  }
+  /* eslint-enable camelcase */
+
+  export default isSelected.extend({
```

Comment:
If you want to use several mixins, you can use the `mixins` attribute, there we need to extend `isSelected` because it's not a mixin but an extended vue object.

I don't remember really well because it's been months I did this but I think extend this `isSelected` object permit TS to know methods of `isSelected`